pallets/ocw-testing/src/lib.rs

```diff
@@ -0,0 +1,215 @@
+            Ok(().into())
+        }
+
+        /// Submits unsigned transaction that emits an event
+        ///
+        /// Can be triggered only by an offchain worker
+        #[pallet::call_index(1)]
+        #[pallet::weight({0})]
+        pub fn submit_event_unsigned(
+            origin: OriginFor<T>,
+            _block_number: BlockNumberFor<T>,
+        ) -> DispatchResultWithPostInfo {
+            // This ensures that the function can only be called via unsigned transaction.
+            ensure_none(origin)?;
+
+            // Increment the block number at which we expect next unsigned transaction.
+            let current_block = <frame_system::Pallet<T>>::block_number();
+
+            // Emits offchain event
+            Self::deposit_event(Event::SimpleOffchainEvent);
+
+            <NextUnsignedAt<T>>::put(current_block + T::UnsignedInterval::get());
+            Ok(().into())
+        }
+    }
+
+    /// Events for the pallet.
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// Simple offchain event
+        SimpleOffchainEvent,
+    }
+
+    #[pallet::validate_unsigned]
+    impl<T: Config> ValidateUnsigned for Pallet<T> {
+        type Call = Call<T>;
+
+        /// Validate unsigned call to this module.
+        ///
+        /// By default unsigned transactions are disallowed, but implementing the validator
+        /// here we make sure that some particular calls (the ones produced by offchain worker)
+        /// are being whitelisted and marked as valid.
+        fn validate_unsigned(_source: TransactionSource, call: &Self::Call) -> TransactionValidity {
+            if let Call::submit_event_unsigned { block_number } = call {
+                Self::validate_transaction_parameters(block_number)
+            } else {
+                InvalidTransaction::Call.into()
+            }
+        }
+    }
+}
+
+impl<T: Config> Pallet<T> {
+    /// A helper function to sign payload and send an unsigned transaction
+    fn send_raw_unsigned_transaction(block_number: BlockNumberFor<T>) -> Result<(), &'static str> {
+        // Make sure transaction can be sent
+        let next_unsigned_at = NextUnsignedAt::<T>::get();
+        if next_unsigned_at > block_number {
+            return Err("Too early to send unsigned transaction");
+        }
+        let is_offchain_worker_enabled = OffchainWorkerTestEnabled::<T>::get();
+        if !is_offchain_worker_enabled {
+            return Err("Offchain worker is not enabled");
+        }
+
+        let call = Call::submit_event_unsigned { block_number };
+
+        SubmitTransaction::<T, Call<T>>::submit_unsigned_transaction(call.into())
+            .map_err(|()| "Unable to submit unsigned transaction.")?;
+
+        Ok(())
+    }
+
+    fn validate_transaction_parameters(block_number: &BlockNumberFor<T>) -> TransactionValidity {
+        // Now let's check if the transaction has any chance to succeed.
```

Comment:
Add check that the testing mode is enabled

Reply:
Added in [2497476](https://github.com/moondance-labs/tanssi/pull/723/commits/2497476aad7fc12254795810cbc4ae078fab99b9)